### PR TITLE
[submitters] Add an option to update the job title on submitters

### PR DESCRIPTION
**Pull request by cbentejac**
## Description

This PR adds an option, `--submitLabel`, to both the scripts (meshroom_batch) and the Meshroom app to set the title of jobs that are submitted. If a job is submitted with `--submitLabel` unset, the default name of the job will be "x [Meshroom]", with x the name of the submitted .mg file.

### CODE REVIEW

File: bin/meshroom_batch [hunk 1]
```diff
@@ -64,6 +64,10 @@ parser.add_argument('--forceCompute', help='Compute in all cases even if already
 
 parser.add_argument('--submit', help='Submit on renderfarm instead of local computation.',
                     action='store_true')
+parser.add_argument("--submitLabel", 
+                    type=str, 
+                    default='meshroom : %j',
```
Comment (fabiencastan) [inline bin/meshroom_batch:69]:
```suggestion
                    default='%j [Meshroom]',
```

File: bin/meshroom_submit [hunk 2]
```diff
@@ -16,7 +16,11 @@ parser.add_argument('--submitter',
                     type=str,
                     default='SimpleFarm',
                     help='Execute job with a specific submitter.')
+parser.add_argument("--submitLabel", 
+                    type=str, 
+                    default='meshroom : %j',
```
Comment (fabiencastan) [inline bin/meshroom_submit:21]:
```suggestion
                    default='%j [Meshroom]',
```

File: meshroom/ui/app.py [hunk 3]
```diff
@@ -134,7 +135,7 @@ def __init__(self, args):
         # instantiate Reconstruction object
         self._undoStack = commands.UndoStack(self)
         self._taskManager = TaskManager(self)
-        r = Reconstruction(undoStack=self._undoStack, taskManager=self._taskManager, defaultPipeline=args.pipeline, parent=self)
+        r = Reconstruction(undoStack=self._undoStack, taskManager=self._taskManager, defaultPipeline=args.pipeline, submitLabel=args.submitLabel, parent=self)
```
Comment (fabiencastan) [inline meshroom/ui/app.py:138]:
```suggestion
        r = Reconstruction(undoStack=self._undoStack, taskManager=self._taskManager, defaultPipeline=args.pipeline, parent=self)
        r.setSubmitLabel(args.submitLabel)
```

File: bin/meshroom_submit (hunk 2 above)
Comment (cbentejac) [reply bin/meshroom_submit:21]: Done

File: bin/meshroom_batch (hunk 1 above)
Comment (cbentejac) [reply bin/meshroom_batch:69]: Done.

File: meshroom/ui/app.py (hunk 3 above)
Comment (cbentejac) [reply meshroom/ui/app.py]: Done.

File: bin/meshroom_batch [hunk 4]
```diff
@@ -64,6 +64,10 @@ parser.add_argument('--forceCompute', help='Compute in all cases even if already
 
 parser.add_argument('--submit', help='Submit on renderfarm instead of local computation.',
                     action='store_true')
+parser.add_argument("--submitLabel", 
+                    type=str, 
+                    default='%j [Meshroom]',
```
Comment (fabiencastan) [inline bin/meshroom_batch:69]: Maybe use the same expression convention used in Meshroom (python format convention):
```suggestion
                    default='{projectName} [Meshroom]',
```

Comment (cbentejac) [reply bin/meshroom_batch]: Done.

**Review by fabiencastan — APPROVED** [review]

File: meshroom/ui/graph.py [hunk 5]
```diff
@@ -261,7 +261,7 @@ class UIGraph(QObject):
     UIGraph exposes undoable methods on its graph and computation in a separate thread.
     It also provides a monitoring of all its computation units (NodeChunks).
     """
-    def __init__(self, undoStack, taskManager, parent=None):
+    def __init__(self, undoStack, taskManager, submitLabel='{projectName}', parent=None):
```
Comment (fabiencastan) [inline meshroom/ui/graph.py:264]: Should be removed from constructor

File: meshroom/ui/graph.py [hunk 6]
```diff
@@ -276,6 +276,7 @@ def __init__(self, undoStack, taskManager, parent=None):
         self._selectedNode = None
         self._selectedNodes = QObjectListModel(parent=self)
         self._hoveredNode = None
+        self.submitLabel = submitLabel
```
Comment (fabiencastan) [inline meshroom/ui/graph.py:279]:
```suggestion
        self.submitLabel = '{projectName}'
```

**Review by fabiencastan — CHANGES_REQUESTED** [review]

File: meshroom/ui/graph.py (hunk 5 above)
Comment (cbentejac) [reply meshroom/ui/graph.py]: Done

File: meshroom/ui/graph.py (hunk 6 above)
Comment (cbentejac) [reply meshroom/ui/graph.py]: Done

**Review by fabiencastan — APPROVED** [review]

**Pull request merged**